geode-core/src/main/java/org/apache/geode/cache/configuration/JndiBindingsType.java

```diff
@@ -237,6 +238,9 @@
     @XmlAttribute(name = "xa-datasource-class")
     protected String xaDatasourceClass;
 
+    @XmlTransient
+    private boolean createdByDataSourceCommand;
```

Comment:
I would prefer not to add this flag here, since it's only used in this specific command. It seems to have dirtied this object. Instead, I would pass this flag as an addition argument to the CreateJndiBindingFunction. See CreateDiskstoreFunction as an example of how to pass multiple arguments to the function.

Reply:
done